ext/standard/array.c

```diff
@@ -6255,40 +6255,19 @@ PHP_FUNCTION(array_map)
    Checks if the given key or index exists in the array */
 PHP_FUNCTION(array_key_exists)
 {
-	zval *key;
+	zend_string *string_key;
+	zend_long int_key;
 	HashTable *ht;
 
 	ZEND_PARSE_PARAMETERS_START(2, 2)
-		Z_PARAM_ZVAL(key)
+		Z_PARAM_STR_OR_LONG(string_key, int_key)
```

Comment:
`array_key_exists()` is not a good target for this. Please see Nikita's comment here: https://github.com/php/php-src/pull/4887/#issuecomment-549327249

Reply:
isset does a usual int silent type cast, string array keys are the only weird thing where "03" will remain "03" but "3" would become the integer index ``3`` thus I think the behaviour would be consistent but can double check